tests/testthat/helper.R

```diff
@@ -1,16 +1,37 @@
 Sys.setlocale("LC_COLLATE", "C") ## What CRAN does
 set.seed(666)
 
-## Our "test package" common harness code
-## `try` added because of some devtools::document weirdness
-try({
-    crunch_test_path <- system.file("crunch-test.R", package="crunch")
-    if (crunch_test_path == "") {
-        # hack for loadall
-        crunch_test_path <- "../../inst/crunch-test.R"
+# find a file starting with the current directory and looking (recursively) in parent folders limited to levels_up
+find_file <- function (file_name, levels_up = 3) {
+    for (i in 0:levels_up) {
+        if (i == 0) {
```

Comment:
The contents of the loop could be simpler:

```
    f <- file.path(paste0(rep("..", i), collapse = .Platform$file.sep), file_name)
    if (file.exists(f)) return(f)
```

Reply:
I might be missing something obvious, but that wouldn't search recursively, right? I've simplified that for loop in other ways, but in order to get into the `inst` directory when we need to, I've set it up to search recursively. I've also limited the search the folder specified with `paste0(rep("..", i)` or `inst` inside of it. This limits the scope of searching and the code is a bit cleaner.

Reply:
That's what goes inside the for loop. So it recurses (starting at cwd) until it finds a file, then breaks and returns it.

Reply:
Without specifying `inst`, that won't actually work:
```
setwd("recrunch/tests/testthat/")

for (i in 0:4) {
    f <- file.path(paste0(rep("..", i), collapse = .Platform$file.sep), "crunch-test.R")
    if (file.exists(f)) print("Found crunch-test.R")
}
``` 
Never finds `test-crunch.R`. Adding `inst` like I did will work, but I also want to make sure if `test-crunch.R` has been moved out of `inst` already, we still catch it. I might be over thinking this, but I want to make sure we're resilient in case there are any (more) changes in `devtools`/`testthat`'s process for running tests.

Reply:
Wait--don't you *not* want to source this file when devtools::load_all is running tests?

So, I think there is some overthinking happening, and not being someone who has any problem with our test setup, I'm somewhat at a disadvantage in reasoning about this, but how about just this:

```
if (file.exists("inst/crunch-test.R")) {
    source("inst/crunch-test.R")
} else if (nchar(system.file("crunch-test.R", package="crunch"))) {
    source(system.file("crunch-test.R", package="crunch"))
}
```

If you've ever installed crunch, the system.file will work.

Reply:
We actually need the file to be `source`d when `devtools::load_all` is called during the testing process, but not be `source`d when `devtools::load_all` (or `crunchdev::load_all` which is a wrapper around the `devtools` version) is called on its own.

Although `system.file("crunch-test.R", package="crunch")` should work by looking at the installed version of Crunch, it doesn't when the package is being loaded with `devtools::load_all`. I've got a minimal package that shows exactly this behavior, if you care to see it, but the behavior is below: The `mintest` package is installed, but `devtools::load_all` at first doesn't find `inst/test-file`, but confusingly if `devtools::load_all` is run again in the same session it does find the file in `inst`, although it's not in the system library, but in the package directory.
```
Kaf:mintest jkeane$ R -e 'print(system.file("test-file", package="mintest"))'
> print(system.file("test-file", package="mintest"))
[1] "/Users/jkeane/Library/R/3.4/library/mintest/test-file"
> 
Kaf:mintest jkeane$ R -e 'load_all("."); load_all(".")'
> load_all("."); load_all(".")
Loading mintest
[1] ""
Loading mintest
[1] "/Users/jkeane/Documents/mintest/inst/test-file"
> 
```
I've dug through the source of `devtools` and can't find exactly where `load_all` is going wrong. FWIW, with `pkgload::load_all()` this issue seems to be resolved. `pkgload::load_all()` is supposed to replace `devtools::load_all()`, and is already used in the master branch of `devtools`. Though I can't find any sort of roadmap for when that will be released. Once that's transitioned to all of this finding code can go away, hopefully for good.
```
Kaf:mintest jkeane$ R -e 'pkgload::load_all(".")'
> pkgload::load_all(".")
Loading mintest
[1] "/Users/jkeane/Documents/mintest/inst/test-file"
```

Reply:
Even more confusingly, the reason that `devtools::test()` works when the helper file has `system.file()` alone is that it apparently (accidentally) loads the helpers twice https://github.com/hadley/devtools/issues/1579